cmake/external/googletest.cmake

```diff
@@ -13,18 +13,14 @@
 # limitations under the License.
 
 include(ExternalProject)
-include(ExternalProjectFlags)
-
-ExternalProject_GitSource(
-  GOOGLETEST_GIT
-  GIT_REPOSITORY "https://github.com/google/googletest.git"
-  GIT_TAG "release-1.8.0"
-)
 
 ExternalProject_Add(
   googletest
 
-  ${GOOGLETEST_GIT}
+  DOWNLOAD_DIR ${PROJECT_BINARY_DIR}/downloads
```

Comment:
This puts the downloaded tarballs into ./build/downloads. Alternatively, we could put them into the top level directory (./downloads or similar). This would prevent them from being deleted when the user make clean's via a `rm -rf ./build`. OTOH, it's kinda surprising that the build would touch files outside of ./build.

Reply:
This LGTM. Travis caching doesn't really care.